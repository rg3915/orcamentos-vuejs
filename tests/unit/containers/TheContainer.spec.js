import CoreuiVue from '@coreui/vue'
import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import TheContainer from '@/containers/TheContainer'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)
localVue.use(CoreuiVue)
const store = new Vuex.Store({
  state: {
    darkMode: false,
    sidebarShow: 'responsive',
    sidebarMinimize: false,
    asideShow: false
  }
})

const router = new VueRouter()

describe('TheContainer.vue', () => {
  it('has a name', () => {
    expect(TheContainer.name).toBe('TheContainer')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(TheContainer, {
      store,
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(TheContainer, {
      store,
      localVue,
      router
    })
    expect(wrapper.vm).toBeTruthy()
  })
})
