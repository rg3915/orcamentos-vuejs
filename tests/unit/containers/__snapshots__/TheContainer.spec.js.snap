// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheContainer.vue renders correctly 1`] = `
<div
  class="c-app"
>
  <thesidebar-stub />
   
  <cwrapper-stub
    tag="div"
  >
    <theheader-stub />
     
    <div
      class="c-body"
    >
      <main
        class="c-main"
      >
        <ccontainer-stub
          fluid="true"
          tag="div"
        >
          <transition-stub
            mode="out-in"
            name="fade"
          >
            <router-view-stub
              name="default"
            />
          </transition-stub>
        </ccontainer-stub>
      </main>
    </div>
     
    <thefooter-stub />
  </cwrapper-stub>
</div>
`;
